--- ckan/ckanext/datastore/backend/postgres.py
+++ ckan/ckanext/datastore/backend/postgres.py
@@ -1809,7 +1809,7 @@ class DatastorePostgresqlBackend(DatastoreBackend):
         read only user.
         '''
         write_connection = self._get_write_engine().connect()
-        read_connection_user = sa_url.make_url(self.read_url).username
+        read_connection_user = sa_url.make_url(self.read_url).username.split("@")[0]
 
         drop_foo_sql = u'DROP TABLE IF EXISTS _foo'
 
