--- ckan/ckanext/datapusher/logic/action.py	2020-09-19 21:25:06.525350202 +0200
+++ action_fix_callback_url.py	2020-09-19 21:27:43.596501963 +0200
@@ -61,13 +61,13 @@
 
     datapusher_url = config.get('ckan.datapusher.url')
 
-    site_url = h.url_for('/', qualified=True)
-
     callback_url_base = config.get('ckan.datapusher.callback_url_base')
     if callback_url_base:
+        site_url = callback_url_base
         callback_url = urlparse.urljoin(
             callback_url_base.rstrip('/'), '/api/3/action/datapusher_hook')
     else:
+        site_url = h.url_for('/', qualified=True)
         callback_url = h.url_for(
             '/api/3/action/datapusher_hook', qualified=True)
 
